options.go

```diff
@@ -91,23 +117,26 @@ func IgnoreCurrent() Option {
 	})
 }
 
```

Comment:
Please add docstring here on what MaxSleep represents, ideally with a code sample on how to use this. 

Re: naming, MaxSleep might not be the best API name. This isn't an issue when this was a private method but if we're exposing it, let's rename this to something more descriptive, such as `MaxRetryInterval`.

Reply:
Sorry for missing that, added.